File: App/src/models/exam/exam.ts
export class Exam {

  name: String;
  date: Date;
  professor: String;
  professorImg: String;
  location: String;
  shareName: String;
  files: Array<String>;

  constructor(name: String = "", date: Date = new Date(Date.now()), professor: String = "", professorImg: String = "",
              location: String = "", shareName: String = "", files: Array<String> = null) {
    this.name = name;
    this.date = date;
    this.professor = professor;
    this.professorImg = professorImg;
    this.location = location;
    this.shareName = shareName;
    this.files = files;
  }
  
}

File: Server/Models/Exam.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Server.Models
{
    public class Exam
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }
        public DateTime Date { get; set; }
        public string Professor { get; set; }
        public string ProfessorImg { get; set; }
        public string Location { get; set; }
        public string ShareName { get; set; }
        public List<File> Files { get; set; }
    }
}
